pkg/server/commands/list_objects.go

```diff
@@ -294,6 +299,7 @@ func (q *ListObjectsQuery) evaluate(
 			}(res)
 		}
 
+		cancel()
 		wg.Wait()
 		close(resultsChan)
```

Comment:
can we put tracing around this code? so that if in the future someone reports an issue about ListObjects being slow, we'll be able to see if time is being spent waiting?

Reply:
Not in this PR, but if you want to create a follow up issue we can consider it. We already have tracing in ReverseExpand and that should give a very good indication if we're blocked waiting on a channel send or if we're sluggish because of database queries being blocked based on concurrency settings etc.. I'm not sure I see the value of adding a span here. In either case, it's outside the scope of the original reason this PR was created.